mobius-extras/src/main/java/com/spotify/mobius/extras/MergedEventSource.java

```diff
@@ -0,0 +1,48 @@
+package com.spotify.mobius.extras;
+
+import com.spotify.mobius.EventSource;
+import com.spotify.mobius.disposables.Disposable;
+import com.spotify.mobius.functions.Consumer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/**
+ * An {@link EventSource} that merges multiple sources into one
+ *
+ * @param <E> The type of Events the sources will emit
+ */
+public class MergedEventSource<E> implements EventSource<E> {
+  private final List<EventSource<E>> eventSources;
+
+  @SafeVarargs
+  public static <E> EventSource<E> from(EventSource<E> source, EventSource<E>... sources) {
+    List<EventSource<E>> allSources = new ArrayList<>();
```

Comment:
Should guard against null event sources somewhere; either here, or in the constructor.